examples/paywall-tester/src/main/java/com/revenuecat/paywallstester/ui/screens/main/paywalls/PaywallsScreen.kt

```diff
@@ -79,19 +87,57 @@ fun PaywallsScreen(
     }
     val currentState = displayPaywallState
     if (currentState is DisplayPaywallState.FullScreen) {
-        PaywallDialog(
-            PaywallDialogOptions.Builder(dismissRequest = {
-                displayPaywallState = DisplayPaywallState.None
+        FullScreenDialog(currentState) {
+            displayPaywallState = DisplayPaywallState.None
+        }
+    } else if (currentState is DisplayPaywallState.Footer) {
+        FooterDialog(currentState) {
+            displayPaywallState = DisplayPaywallState.None
+        }
+    }
+}
+
+@Composable
+private fun FullScreenDialog(currentState: DisplayPaywallState.FullScreen, onDismiss: () -> Unit) {
+    PaywallDialog(
+        PaywallDialogOptions.Builder(dismissRequest = onDismiss)
+            .setOffering(currentState.offering)
+            .setListener(object : PaywallListener {
+                override fun onPurchaseCompleted(customerInfo: CustomerInfo, storeTransaction: StoreTransaction) {
+                    onDismiss()
+                }
             })
-                .setOffering(currentState.offering)
-                .setListener(object : PaywallListener {
-                    override fun onPurchaseCompleted(customerInfo: CustomerInfo, storeTransaction: StoreTransaction) {
-                        displayPaywallState = DisplayPaywallState.None
-                    }
-                })
-                .setFontProvider(currentState.fontProvider)
-                .build(),
-        )
+            .setFontProvider(currentState.fontProvider)
+            .build(),
+    )
+}
+
+@Composable
+private fun FooterDialog(currentState: DisplayPaywallState.Footer, onDismiss: () -> Unit) {
+    Dialog(
+        onDismissRequest = onDismiss,
+        properties = DialogProperties(usePlatformDefaultWidth = false),
+    ) {
+        Scaffold { scaffoldPadding ->
+            Box(modifier = Modifier.padding(scaffoldPadding)) {
+                PaywallFooter(
+                    options = PaywallOptions.Builder()
+                        .setOffering(currentState.offering)
+                        .setListener(object : PaywallListener {
```

Comment:
This is repeated now. Can we just create an actual private class of `PaywallListener` (that takes an `onDismiss`) to use it in all of these?